### ISSUE TITLE
Map doesn't load from RaspberryPi

### ISSUE THREAD

**Issue by krummrey** [PokemonGoMap/PokemonGo-Map#906]
**I'm trying to set up the map on my Rasberry Pi.
However when I try to access the Map it loads the page but the JavaScript map fails to load. 
I just get the HTML.
This is the start of the outtput on the console:**

2016-08-21 12:52:54,778 [      MainThread][     runserver][    INFO] Parsed location is: XX.XX/XX.XX (lat/lng/alt)
2016-08-21 12:52:54,853 [      MainThread][        models][    INFO] Connecting to MySQL database on 127.0.0.1:3306
2016-08-21 12:52:55,036 [ search-overseer][        search][    INFO] Search overseer starting
2016-08-21 12:52:55,048 [ search-overseer][        search][    INFO] Starting search worker threads
2016-08-21 12:52:55,090 [ search-overseer][        search][    INFO] New location caught, moving search grid
2016-08-21 12:52:55,098 [ search-overseer][        search][    INFO] Scan Distance is 0.07 km
2016-08-21 12:52:55,170 [      db-cleaner][        models][    INFO] Regular database cleaning complete
2016-08-21 12:52:55,203 [ search-worker-0][        search][    INFO] Search step 1 beginning (queue size is 0)
2016-08-21 12:52:55,214 [ search-worker-0][      auth_ptc][    INFO] PTC User Login for: XXX
2016-08-21 12:52:57,353 [ search-worker-0][      auth_ptc][    INFO] PTC User Login successful.
2016-08-21 12:52:57,367 [ search-worker-0][      auth_ptc][    INFO] Request PTC Access Token...
2016-08-21 12:52:57,629 [ search-worker-0][      auth_ptc][    INFO] PTC Access Token successfully retrieved.
2016-08-21 12:53:00,466 [ search-worker-0][        models][    INFO] Parsing found 1 pokemons, 5 pokestops, and 0 gyms
2016-08-21 12:53:04,480 [ search-worker-0][        search][    INFO] Search step 2 beginning (queue size is 35)
...

**The output when I load the map in a browser on the console is:**
[2016-08-21 13:02:09,510] ERROR in app: Exception on /static/a9df1cf/js/vendor/classie.js [GET]
Traceback (most recent call last):
  File "/opt/python/lib/python2.7/site-packages/flask/app.py", line 1988, in wsgi_app
    response = self.full_dispatch_request()
  File "/opt/python/lib/python2.7/site-packages/flask/app.py", line 1641, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/opt/python/lib/python2.7/site-packages/flask/app.py", line 1544, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/opt/python/lib/python2.7/site-packages/flask/app.py", line 1637, in full_dispatch_request
    rv = self.preprocess_request()
  File "/opt/python/lib/python2.7/site-packages/flask/app.py", line 1837, in preprocess_request
    rv = func()
  File "/opt/python/lib/python2.7/site-packages/playhouse/flask_utils.py", line 165, in connect_db
    self.database.connect()
  File "/opt/python/lib/python2.7/site-packages/peewee.py", line 3402, in connect
    **self.connect_kwargs)
  File "/opt/python/lib/python2.7/site-packages/playhouse/pool.py", line 127, in _connect
    raise ValueError('Exceeded maximum connections.')
ValueError: Exceeded maximum connections.

**So flask is the problem? The install seemed fine except for this line:**
Collecting flask_cachebust from git+https://github.com/ChrisTM/Flask-CacheBust.git@aa09ad861f104f987928fe9f5107ccfe0e4473c3#egg=flask_cachebust (from -r requirements.txt (line 26))
  Cloning https://github.com/ChrisTM/Flask-CacheBust.git (to aa09ad861f104f987928fe9f5107ccfe0e4473c3) to /tmp/pip-build-faU1OO/flask-cachebust
  Could not find a tag or branch 'aa09ad861f104f987928fe9f5107ccfe0e4473c3', assuming commit.

**In the end everything seemed fine:**
Successfully installed Flask-0.11.1 Jinja2-2.8 LatLon-1.0.1 MarkupSafe-0.23 PyMySQL-0.7.5 PySocks-1.5.6 PyYAML-3.11 Pygments-2.1.3 Werkzeug-0.11.10 alabaster-0.7.9 argh-0.26.2 babel-2.3.4 backports-abc-0.4 certifi-2016.8.8 click-6.6 commonmark-0.5.4 configargparse-0.10.0 docutils-0.12 flask-cachebust-1.0.0 flask-compress-1.3.0 flask-cors-2.1.2 future-0.15.2 geopy-1.11.0 gpsoauth-0.3.0 imagesize-0.7.1 itsdangerous-0.24 livereload-2.4.1 pathtools-0.1.2 peewee-2.8.1 pgoapi-1.1.6 port-for-0.3.1 protobuf-3.0.0 protobuf-to-dict-0.1.0 pycryptodomex-3.4.2 pyproj-1.9.5.1 pytz-2016.6.1 recommonmark-0.4.0 requests-2.10.0 s2sphere-0.2.4 setuptools-26.0.0 singledispatch-3.4.0.3 six-1.10.0 snowballstemmer-1.2.1 sphinx-1.4.5 sphinx-autobuild-0.6.0 sphinx-rtd-theme-0.1.9 tornado-4.4.1 watchdog-0.8.3 xxhash-0.6.1

**Linux raspi 4.4.18+ #905 Fri Aug 19 19:04:48 BST 2016 armv6l GNU/Linux
Python 2.7.12
pip 8.1.2 from /opt/python/lib/python2.7/site-packages (python 2.7)
Node v6.4.0
npm 3.10.6**

**Comment by angrymoose** [comment]
It sounds like an issue with your particular setup rather than a issue with the project. With that said what linux distro are you running, linux raspi just gets me a list of ditros that run on raspberry pi's. Also for a quick test try the link below to upgrade your install, it tends to fix a lot of problems.

https://pgm.readthedocs.io/en/develop/basic-install/index.html#updating-the-application

**Comment by krummrey** [comment]
**I've done the steps described there and got this at the end:**

npm ERR! Failed at the pogomap@0.1.0 build script 'grunt build'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the pogomap package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     grunt build

0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/node', '/usr/local/bin/npm', 'run', 'build' ]
2 info using npm@3.10.6
3 info using node@v6.4.0
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle pogomap@0.1.0~prebuild: pogomap@0.1.0
6 silly lifecycle pogomap@0.1.0~prebuild: no script for prebuild, continuing
7 info lifecycle pogomap@0.1.0~build: pogomap@0.1.0
8 verbose lifecycle pogomap@0.1.0~build: unsafe-perm in lifecycle true
9 verbose lifecycle pogomap@0.1.0~build: PATH: /usr/local/lib/node_modules/npm/bin/node-gyp-bin:/home/pi/Pokemon/PokemonGo-Map/node_modules/.bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
10 verbose lifecycle pogomap@0.1.0~build: CWD: /home/pi/Pokemon/PokemonGo-Map
11 silly lifecycle pogomap@0.1.0~build: Args: [ '-c', 'grunt build' ]
12 silly lifecycle pogomap@0.1.0~build: Returned: code: 3  signal: null
13 info lifecycle pogomap@0.1.0~build: Failed to exec build script
14 verbose stack Error: pogomap@0.1.0 build: `grunt build`
14 verbose stack Exit status 3
14 verbose stack     at EventEmitter.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/lifecycle.js:255:16)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at EventEmitter.emit (events.js:191:7)
14 verbose stack     at ChildProcess.<anonymous> (/usr/local/lib/node_modules/npm/lib/utils/spawn.js:40:14)
14 verbose stack     at emitTwo (events.js:106:13)
14 verbose stack     at ChildProcess.emit (events.js:191:7)
14 verbose stack     at maybeClose (internal/child_process.js:852:16)
14 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:5)
15 verbose pkgid pogomap@0.1.0
16 verbose cwd /home/pi/Pokemon/PokemonGo-Map
17 error Linux 4.4.18+
18 error argv "/usr/local/bin/node" "/usr/local/bin/npm" "run" "build"
19 error node v6.4.0
20 error npm  v3.10.6
21 error code ELIFECYCLE
22 error pogomap@0.1.0 build: `grunt build`
22 error Exit status 3

**Comment by drewreece** [comment]
The Pi needs to have node-sass rebuilt to work - I don't think there is an arm version of it available, this seemed to get it rebuilt for me… 
npm rebuild node-sass

Before I found that tip I used this…
./node_modules/.bin/grunt build --force
& then manually examined the failing webpage (view source) & copied over relevant js & css files from a release version to the places that were 404'ing.

Frankly running the release version is best on a Pi - it takes eons to install node & other dependancies & you get little benefit from the bleeding edge compared to how often you need to rebuild etc.

I don't know if node-sass also builds js but it is clear you are failing to build the static files via the develop toolchain.

**Comment by b3rguinho** [comment]
on my Pi works fine (Raspbian)
with node 4.4.7 and pip 8.1.2

**Comment by krummrey** [comment]
@drewreece Using the release version did the trick. Should have thought of that. Will stick to the releases then. Thanks

**Comment by cyan** [comment]
@b3rguinho how did you got those version installed? i have some that are like centuries ago TT thx

**Comment by drewreece** [comment]
Cyan, the Pi is Arm architecture (v6,v7 and v8). Packages are slow to be built & can lag behind if they are ever compiled for it - some libraries are not supported or are incompatible etc.

I have pip 8.1.2, node v0.10.29, npm 1.4.21  (via pip -v, node -v, npm -v) on the Pi and that is current in the repositories as far as I can tell. I'm not sure if we should be compiling newer versions ourselves as it seems OK so far.

I don't understand how node can be out of date for @b3rguinho, isn't 4.4.7 from the future or did the version numbering change? Maybe those numbers are from the package & not the actual release?

**Comment by cyan** [comment]
https://nodejs.org/dist/ i tried the 4.4

**Comment by drewreece** [comment]
@cyan, all I can say is that the develop branch works for me on a Pi model B (rev 2 if I recall correctly).

I have packages from apt-get as the old develop setup indicates. I followed the original guide a while ago…
https://github.com/PokemonGoMap/PokemonGo-Map/wiki/Development

As I detailed earlier node-sass failed until I did a npm rebuild node-sass (probably with sudo). It has been fine since then.

It seems like your issue is in your setup, sorry I fear you are going to be told that 'github is not for support' as is this project's policy.

Take it to discord for help or troubleshooting as the new docs lists that for support…https://pgm.readthedocs.io/en/develop/
https://discord.gg/PWp2bAm

P.S. it would help anyone in discord if you told them what your actual error is so far we know nothing of what you are seeing.
